### PR TITLE
Caporal and other nice things

### PR DESCRIPTION
**Pull request by hannah-bulmer**
- move colours to chalk and expose them in a better way
- get rid of callbacks in favour of some cleaner async/await
- move multiple CLI tools together into one
- migrate commander to caporal
- update readme

### CODE REVIEW

File: lib/theme.js [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
+  else if (level === 'warn') console.log(chalk.yellow(msg));
+  else if (level === 'error') console.log(chalk.red(msg));
+  else if (level === 'data') console.log(chalk.magenta(msg));
+  else console.log(msg);
+}
+
+exports.ERROR = 'error';
+exports.INFO = 'info';
+exports.WARN = 'warn';
+exports.DATA = 'data';
```
Comment (dvd-z) [inline lib/theme.js]: it's a bit annoying to have to import individual log levels. maybe have a Map/Object that acts as a levels enum

```
exports.Levels = Object.freeze({
  ERROR: 'error',
  INFO: 'info',
  WARN: 'warn',
  DATA: 'data',
});
```

File: lib/theme.js [hunk 2]
```diff
@@ -1,7 +1,15 @@
-const colors = require('colors');
+/* eslint-disable no-console */
+const chalk = require('chalk');
 
-colors.setTheme({
-  info: 'green',
-  warn: 'yellow',
-  error: 'red',
-});
+exports.print = function print(msg, level) {
```
Comment (dvd-z) [inline lib/theme.js]: also I think it's more clear to have level be the first parameter

File: lib/theme.js (hunk 1 above)
Comment (hannah-bulmer) [reply lib/theme.js]: Learned about Object.freeze today

File: lib/theme.js (hunk 2 above)
Comment (hannah-bulmer) [reply lib/theme.js]: Discussed offline :)

File: lib/process.js [hunk 3]
```diff
@@ -1,71 +1,28 @@
...
-    .on('end', onEnd);
 
   try {
     fn.apply(scope, [msg, cfg, snapshot]);
     helpers.destroyProcess();
   } catch (e) {
-    console.log(e.stack);
+    print.error(e.stack);
     helpers.destroyProcess();
   }
```
Comment (dvd-z) [inline lib/process.js]: remove the repeated `helpers.destroyProcess()` and add a finally block
```
finally {
  helpers.destroyProcess();
}

File: lib/prompt.js [hunk 4]
```diff
@@ -1,73 +1,61 @@
...
-    const result = fixtures[key];
-    if (!result) {
-      console.log("Fixture '%s' does not exist".error, key);
-      helpers.destroyProcess();
-      cb();
-    }
-    cb(result);
-  };
+  if (!fixtures) {
+    print('No fixtures found', ERROR);
```
Comment (dvd-z) [inline lib/prompt.js]: missed this one

File: lib/prompt.js [hunk 5]
```diff
@@ -1,73 +1,61 @@
...
+    return;
```
Comment (dvd-z) [inline lib/prompt.js]: perhaps better to throw an error since `retrieveFixture` is awaited and expects a response

File: lib/prompt.js [hunk 6]
```diff
@@ -1,73 +1,61 @@
...
-  } else {
-    findFunctionByName(functionName);
   }
+
+  const result = component[functionName];
+
+  if (!result) {
+    print.error(`Function '${functionName}' does not exist`);
+    helpers.destroyProcess();
+    return;
```
Comment (dvd-z) [inline lib/prompt.js]: same

**Pull request merged**